Update TS ThunkAction types and returned values in action creators

migmartri opened the pull request:
It updates the ReduxThunk action creators signature types. It also removes unnecessary returned values since an action creator should just do that, dispatch actions, not to be consumed directly. There are still a couple of actions returning a boolean that we could change later. 

Closes https://github.com/kubeapps/kubeapps/issues/658

prydonius commented on dashboard/src/actions/apps.ts:
```diff
@@ -55,8 +54,11 @@ const allActions = [
+export function getApp(
+  releaseName: string,
+  namespace: string,
+): ThunkAction<Promise<void>, IStoreState, null, AppsAction> {
```
why don't we alias this type so it's easier to reuse? e.g. we could create a `KubeappsThunkAction<void, AppsAction>` type, the first parameter being the return of the promise and the last being the action type.

andresmgot replied:
I think it's a bit easier to understand if we leave this as `ThunkAction`. 

For example, reviewing this PR I needed to investigate what the parameters of `ThunkAction` meant. If we add `KubeappsThunkAction` first I would need to check what the parameters mean there and then how it's forwarded to `ThunkAction` and finally check the implementation of that. I would avoid that extra step if it's just to avoid typing the two parameters `IStoreState, null`.

migmartri replied:
I do not have strong opinions, but I agree with @andresmgot that this thing is convoluted and complex enough that adding another layer of indirection will not help.

prydonius replied:
yeah I agree that it's not that important, thanks!

prydonius commented on dashboard/src/actions/auth.ts:
```diff
@@ -27,22 +28,24 @@ const allActions = [setAuthenticated, authenticating, authenticationError];
       await Auth.validateToken(token);
       Auth.setAuthToken(token);
-      return dispatch(setAuthenticated(true));
+      dispatch(setAuthenticated(true));
```
I think the reason I decided to return this is because actions created by `createAction` also returned the action, so it was to have the same behaviour, but we are not really using it so I don't mind making this change.

migmartri replied:
They are different things no?

createAction returns an action creator that then you need to `dispatch` no? In this case we are returning a dispatched value?

prydonius commented on dashboard/src/components/AppUpgrade/AppUpgrade.tsx:
```diff
@@ -29,11 +29,11 @@ interface IAppUpgradeProps {
-  getBindings: () => Promise<IServiceBindingWithSecret[]>;
-  getChartVersion: (id: string, chartVersion: string) => Promise<{}>;
-  getChartValues: (id: string, chartVersion: string) => Promise<any>;
+  fetchChartVersions: (id: string) => Promise<any>;
```
shouldn't this be void also?

andresmgot replied:
`fetchChartVersions` returns a list of `IChartVersions` and it's used in `src/actions/charts.ts`

migmartri replied:
Avoid what? The use of any? And what's the difference of returning an empty object.

To be clear, what I am trying to do here is to be sure that we use action creations to only dispatch values, not to rely on any internal returned value and use them outside of the redux scope. 

In this specific case though I believe that I could not remove the returned value because as @andresmgot mentioned, this function is called from another action creator and changing it will require a bigger change.

So basically I've tried to not to return anything for actions that are pure but maintain for now the others.

In any case, I am going to double check this one, thanks for pointing it out.

prydonius replied:
> Avoid what? The use of any? And what's the difference of returning an empty object.

I think you misread, I was asking why this isn't `void`, but I see now that the return value is being used and we don't want to change this.

migmartri replied:
Ahhh, sorry for the misunderstanding. Yes. that's the reason.

migmartri commented:
@andresmgot, @prydonius, thank you for your reviews.

Based on your comments I have changed a couple of things:

* For the function props passed to the components, I have properly typed them (many of them had => `:any`), thanks @andresmgot for the `Promise<IChartVersion[]>;` catch.
* For the ones that the action creator returned `Promise<void>` I have changed their type in the component to `void` directly, that way we basically are saying that the consumer should not do anything (async/await) with those. Thanks @prydonius for that suggestion.
* This patch does not change any returned value for those action creations, although in future iterations we should probably start refactoring them to in general not to depend of their returned value but instead use the global redux store dispatching transient events (i.e deploying)
